### DISCUSSION TITLE
kubectl 명령어 자동완성 기능 사용 어떻게?? 하나요?

### DISCUSSION THREAD

**Discussion by hyunkyung13306** [yoonjeong-kwon/fastcampus-kubernetes#3] [Q&A] [answered]
안녕하세요~~ 강의 잘 듣고 있습니다.

part3.ch1.03 에는 자동 완성 설치는 안하신거 같은데...

kebectl 명령어 자동 완성 설치 어떻게 하나요??

https://kubernetes.io/ko/docs/tasks/tools/install-kubectl-linux/
해당 부분 참조 했는데 안되요.....

apt-get install bash-completion
E: Conflicting values set for option Signed-By regarding source https://packages.cloud.google.com/apt/ cloud-sdk: /usr/share/keyrings/cloud.google.gpg !=
E: The list of sources could not be read.
E: Conflicting values set for option Signed-By regarding source https://packages.cloud.google.com/apt/ cloud-sdk: /usr/share/keyrings/cloud.google.gpg !=
E: The list of sources could not be read.

window10+구글 gke 사용+sdk+ vscode(터미널 ubuntu로 ) 
구글 검색해도 방법을 모르겠어요~~~

**Comment by yoonjeong-kwon** [comment, marked as answer]
@hyunkyung13306 님 안녕하세요. 반갑습니다 😊

사용하고 계신 OS는 윈도우이신가요? 아니면 리눅스이신가요?
참조하신 링크는 리눅스용 kubectl 명령어 설치 방법이고 검색하신 키워드는 윈도우라서요.

kubectl 명령어 실행과 자동 완성 기능은 다르기 때문에 나누어 설명드리고자 합니다.
먼저 kubectl 명령어를 이용해 쿠버네티스 리소스 제어를 하시려면
https://kubernetes.io/ko/docs/tasks/tools/ 에서 OS에 맞는 메뉴를 선택해서 설치해주시면 되겠습니다.

그 다음 kubectl 명령어 자동 완성은 쉘에서 지원하는 기능인데요.
제가 실습 영상에 사용한 환경을 설명드릴게요.
저는 맥OS를 사용하고 있고 맥OS에는 zsh이 기본 쉘로 설정되어 있는데요. 
zsh에 플러그인으로 자동완성 기능이 포함되어 있거든요.
zsh 플러그인으로는 `zsh-completions`, `zsh-autosuggestions`을 사용하고 있습니다.

제가 윈도우 설치 방법을 잘 몰라서 zsh 설치에 도움될만한 몇가지 링크를 찾아 첨부드립니다. 
- [Windows10에 Oh My Zsh 설치하는 방법(영문)](https://dev.to/vsalbuq/how-to-install-oh-my-zsh-on-windows-10-home-edition-49g2)
  - 여기서 7번 플러그인(`zsh-completions`, `zsh-autosuggestions`)을 설치해주시면 됩니다.
- [Oh My Zsh 공식 홈페이지](https://ohmyz.sh/)
- [Oh My Zsh Github 위키](https://github.com/ohmyzsh/ohmyzsh/wiki)

아래 링크는 kubectl 자동 완성 스크립트와 관련된 kubernetes.io 가이드 내용입니다.
이것을 먼저 시도 해보시면 문제를 간단하게 해결할 수도 있을 것 같아요.
- [PowerShell 자동 완성
](https://kubernetes.io/ko/docs/tasks/tools/included/optional-kubectl-configs-pwsh/)
- [리눅스에서 kubectl 자동 완성 활성화](https://kubernetes.io/ko/docs/tasks/tools/included/optional-kubectl-configs-bash-linux/) 

또 문의사항 있으시면 질문 남겨주세요. 감사합니다 :)
